velox/exec/HybridExecOperator.cpp

```
@@ -33,33 +33,216 @@ Operator::PlanNodeTranslator HybridExecOperator::planNodeTranslator =
+  } else if (isFilter_) {
+    return tmpOut;
+  }
+
   return std::move(result_);
 }
 
 void HybridExecOperator::process() {
-  // 1. convert data
-  // Convertor.V2O
+  if (isGroupBy_) {
+    throw std::runtime_error("not support now!");
+  }
+  if (isAgg_) {
+    // 1. convert input data.
+    int64_t numRows = input_->size();
+    auto ciderBuffer = dataConvertor_->convertToCider(input_, numRows);
+
+    int32_t colNum = partialAggResult_.size(); // FIXME:
+    int64_t** outBuffers = (int64_t**)std::malloc(sizeof(int64_t*) * colNum);
+    outBuffers[0] =
+        (int64_t*)std::malloc(sizeof(int64_t) * 1); // agg only return 1 row
 
-  // 2. call ciderKernel_ run
-  // ciderKernel_->runWithData();
+    int32_t matchedRows = 0;
+    int32_t errCode = 0;
 
-  // 3. convert data
-  // result_ = Convertor.O2V()
+    // 2. convert partial result
+    int64_t* aggInitValue =
+        partialAggResult_.data(); // TODO: verify, this will be overwrite?
 
+    // 3. call run method
+    ciderKernel_->runWithData(
+        (const int8_t**)ciderBuffer.colBuffer,
+        &numRows,
+        outBuffers,
+        &matchedRows,
+        &errCode,
+        aggInitValue);
+
+    // 4. convert result into partial agg result
+    partialAggResult_[0] = outBuffers[0][0];
+  } else if (isFilter_) {
+    // NOTE: not ready yet!!
+    // 1. convert input data.
+    int64_t numRows = input_->size();
+    auto ciderBuffer = dataConvertor_->convertToCider(input_, numRows);
+
+    int32_t colNum = 1; // FIXME:
+    int64_t** outBuffers = (int64_t**)std::malloc(sizeof(int64_t*) * colNum);
+
+    int32_t matchedRows = 0;
+    int32_t errCode = 0;
+
+    // 2. call run method
+    ciderKernel_->runWithData(
+        (const int8_t**)ciderBuffer.colBuffer,
+        &numRows,
+        outBuffers,
+        &matchedRows,
+        &errCode,
+        nullptr);
+
+    // 3. convert result into tmpOut/ RowVectorPtr
+    // TODO: seems a lot overhead here
+  }
+
+  if (isSort_) {
+    // TODO: impl
```

For now, let's just throw an exception.